x-pack/legacy/plugins/apm/public/utils/formatters.ts

```diff
@@ -245,3 +249,145 @@ export const getFixedByteFormatter = memoize((max: number) => {
+    case 'milliseconds':
+      return 'HH:mm:ss.SSS';
+    default:
+      return '';
+  }
+}
+
+type DatePrecision = 'days' | 'months' | 'years';
+function getDateFormat(precision: DatePrecision) {
+  switch (precision) {
+    case 'years':
+      return 'YYYY';
+    case 'months':
+      return 'MMM YYYY';
+    case 'days':
+      return 'MMM D, YYYY';
+    default:
+      return '';
+  }
+}
+
+function getFormatsAccordingToDateDifference(
+  momentStart: moment.Moment,
+  momentEnd: moment.Moment
+) {
+  const getDateDifference = (unitOfTime: DatePrecision | TimePrecision) =>
+    momentEnd.diff(momentStart, unitOfTime);
+
+  // Difference is greater than or equals to 5 years
+  if (getDateDifference('years') >= 5) {
+    return { dateFormat: getDateFormat('years') };
+  }
+  // Difference is greater than or equals to 5 months
+  if (getDateDifference('months') >= 5) {
+    return { dateFormat: getDateFormat('months') };
+  }
+
+  const dateFormat = getDateFormat('days');
+  // Difference is greater than 1 day
+  if (getDateDifference('days') > 1) {
+    return { dateFormat };
+  }
+
+  // Difference is greater than or equals to 5 hours
+  if (getDateDifference('hours') >= 5) {
+    return { dateFormat, timeFormat: getTimeFormat('minutes') };
+  }
+
+  // Difference is greater than or equals to 5 minutes
+  if (getDateDifference('minutes') >= 5) {
+    return { dateFormat, timeFormat: getTimeFormat('seconds') };
+  }
+
+  // When difference is in milliseconds
+  return { dateFormat, timeFormat: getTimeFormat('milliseconds') };
+}
+
+export function asAbsoluteTime({
+  time,
+  precision = 'milliseconds'
+}: {
+  time: number;
+  precision?: TimePrecision;
+}) {
+  const momentTime = moment(time);
+  const formattedTz = formatTimezone(momentTime);
+
+  return momentTime.format(
+    `${getDateFormat('days')}, ${getTimeFormat(precision)} ${formattedTz}`
+  );
+}
+
+/**
+ *
+ * Returns the dates formatted according to the difference between the two dates:
+ *
+ * | Difference     |           Format                               |
+ * | -------------- |:----------------------------------------------:|
+ * | >= 5 years     | YYYY - YYYY                                    |
+ * | >= 5 months    | MMM YYYY - MMM YYYY                            |
+ * | > 1 day        | MMM D, YYYY                                    |
+ * | >= 5 hours     | MMM D, YYYY, HH:mm - HH:mm (UTC)               |
+ * | >= 5 minutes   | MMM D, YYYY, HH:mm:ss - HH:mm:ss (UTC)         |
+ * | default        | MMM D, YYYY, HH:mm:ss.SSS - HH:mm:ss.SSS (UTC) |
+ *
+ * @param start timestamp
+ * @param end timestamp
+ */
+export function asRelativeDateRange(start: number, end: number) {
+  const momentStartTime = moment(start);
+  const momentEndTime = moment(end);
+
+  const { dateFormat, timeFormat } = getFormatsAccordingToDateDifference(
+    momentStartTime,
+    momentEndTime
+  );
+
+  if (timeFormat) {
+    const formattedStartTime = momentStartTime.format(
```

Comment:
nit: "formattedStartTime" sounds like it's just the "time" (and thus not the "date").
What do you think about either calling it "formattedStart" or "formattedStartDateTime"

Reply:
Could also be: "startFormatted" or "startDateTimeFormatted"